subgraphs.py

```diff
@@ -127,19 +234,83 @@ def make_subgraphs_per_type(self,sub_type):
         if count > min_subgraph_size:
             mean = current / count
             self.avg_embeddings.append(mean)
-            columnsSquareDiff = np.zeros(self.args.ncomp, dtype=np.float64)
-            for se in similar_entities:
-                columnsSquareDiff += (E[se] - mean) * (E[se] - mean)
-            if count > 2:
-                columnsSquareDiff /= (count-1)
-            else:
-                columnsSquareDiff = mean
-            self.var_embeddings.append(columnsSquareDiff)
+            self.var_embeddings.append(self.calculate_var_embeddings(count, mean, similar_entities))
             self.add_subgraphs(subType, prevo, prevp, count, similar_entities)
 
         print ("# of subgraphs ({}) : {}".format(sub_type_to_string[sub_type], self.get_Nsubgraphs()))
 
-    def make_subgraphs(self):
+    def make_diamond_subgraphs(self, sub_type, adj_list_in, adj_list_out):
+        E = self.E
```

Comment:
This logic can perhaps be improved (time complexity wise)
But let's do that later.
First we will concentrate on correct working of the logic